cmd/agent/app/server.go

```diff
@@ -133,6 +135,31 @@ func (a *Agent) runProxyConnection(o *options.GrpcProxyAgentOptions, drainCh, st
 		}),
 	}
 	cc := o.ClientSetConfig(dialOptions...)
+
+	if o.ServerLeaseSelector != "" {
+		var k8sClient *kubernetes.Clientset
+		config, err := clientcmd.BuildConfigFromFlags("", o.KubeconfigPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load kubernetes client config: %v", err)
+		}
+		k8sClient, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create kubernetes clientset: %v", err)
+		}
+		serverLeaseSelector, err := labels.Parse(o.ServerLeaseSelector)
+		if err != nil {
+			return nil, fmt.Errorf("invalid server count lease selector: %w", err)
+		}
+		serverLeaseCounter := agent.NewServerLeaseCounter(
+			k8sClient,
+			serverLeaseSelector,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create server lease counter: %w", err)
```

Comment:
We should make sure this gets logged. Worried this is opening an failure scenario where KAS cannot call into the cluster; wedging the cluster.

Reply:
Added a klog expression.